altos-rust/altos-core/src/task/control.rs

```diff
@@ -66,34 +66,115 @@ pub enum Priority {
 impl Priority {
     /// Returns a range of values corresponding to each priority from highest to lowest inclusively.
     ///
-    /// Use this to itterate over all priority queues.
+    /// Use this to iterate over all priority queues.
     pub fn get_range_all() -> ::core::ops::Range<usize> {
         (0..NUM_PRIORITIES)
     }
 
     /// Returns a range of values corresponding to each priority from `start` to `end` exclusively.
     ///
-    /// Use this to itterate over priority queues when some at the queues at the beginning and or end
-    /// need to be skipped
-    pub fn get_range(start:Priority, end:Priority) -> ::core::ops::Range<usize> {
+    /// Use this to iterate over priority queues when some at the queues at the beginning and or
+    /// end need to be skipped.
+    pub fn get_range(start: Priority, end: Priority) -> ::core::ops::Range<usize> {
         start as usize .. end as usize
     }
 
     /// Returns a range of values corresponding to each priority from `start` to lowest inclusively.
     ///
-    /// Use this to itterate over priority queues when higher queues have already been iterated over
-    /// or higher priority queues need to be skipped
-    pub fn get_range_to_end(start:Priority) -> ::core::ops::Range<usize> {
+    /// Use this to itterate over priority queues when higher queues have already been iterated
+    /// over or higher priority queues need to be skipped.
+    pub fn get_range_to_end(start: Priority) -> ::core::ops::Range<usize> {
         start as usize .. NUM_PRIORITIES
     }
 
     /// Returns a range of values corresponding to priorities equal to or higher than `self`.
     ///
-    /// Use this to iterate over priority queues starting from the highest priority queue down to the
-    /// current priority.
+    /// Use this to iterate over priority queues starting from the highest priority queue
+    /// down to the current priority.
     pub fn higher(&self) -> ::core::ops::Range<usize> {
```

Comment:
This one is still used in the `sleep` function I believe, but it should also probably be replaced by an iterator